File: Lyft Backend Software Engineering/battery/battery_type/nubbin_battery.py
# Importing the Battery class
from battery.battery import Battery

# NubbinBattery class inheriting from Battery
class NubbinBattery(Battery):
    # Constructor to initialize the current and last service dates
    def __init__(self, current_date, last_service_date):
        self.current_date = current_date
        self.last_service_date = last_service_date
        self.nubbin_year_need_service = 3  # Service needed every 3 years

    # Method to check if the battery needs service
    def needs_service(self):
        # Calculating the date when service is needed
        date_need_service = self.last_service_date.replace(year=self.last_service_date.year + self.nubbin_year_need_service)
        # Returning True if service is needed, otherwise False
        return date_need_service < self.current_date

# Adding a reference to the license and the author
# License: Nikhil Gurjar

File: Softstraw Back-End Engineer/battery/battery_type/nubbin_battery.java
import java.util.Date;

public class NubbinBattery extends Battery {
    // Fields to store current and last service dates
    private Date currentDate;
    private Date lastServiceDate;
    private int nubbinYearNeedService = 3;  // Service needed every 3 years

    // Constructor to initialize the current and last service dates
    public NubbinBattery(Date currentDate, Date lastServiceDate) {
        this.currentDate = currentDate;
        this.lastServiceDate = lastServiceDate;
    }

    // Method to check if the battery needs service
    public boolean needsService() 
    {
        // Calculating the date when service is needed
        int year = lastServiceDate.getYear() + nubbinYearNeedService;
        Date dateNeedService = new Date(year, lastServiceDate.getMonth(), lastServiceDate.getDay());

        // Returning true if service is needed, otherwise false
        return dateNeedService.before(currentDate);
    }

}


# Adding a reference to the license and the author
# License: Nikhil Gurjar
